### DISCUSSION TITLE
Switching from stay armed to night armed

### DISCUSSION THREAD

**Discussion by bourgeois** [taligentx/dscKeybusInterface#126] [dscKeybusInterface Discussion]
I'm trying to write a piece of code that would allow me to change between stay armed and night armed (I want an automation that runs when we are going to bed). I tried this but I'm getting a "System is unavailable" on my other keypads for few seconds and then it doesn't seem to work.... Am I missing something (any global boolean flag I need to change) ?

    // homebridge-mqttthing NIGHT_ARM from STAY_ARM
    if (payload[payloadIndex] == 'N' && 
       dsc.armedStay[partition] && 
      !dsc.exitDelay[partition] && 
      !dsc.noEntryDelay[partition]) {

        dsc.writePartition = partition + 1;    // Sets writes to the partition number
        dsc.write("*9");  // Keypad arm with no entry delay
        publishState(mqttPartitionTopic, partition, "N", 0);
        exitState = 'N';
        return;

    }

I added this code right after the `// Checks if a partition number 1-8 has been sent` bloc.

Am I doing this right ? Would it better to handle this change at the library code level and not the sketch ?

**Comment by duanemck** [comment]
@bourgeois Did you figure this out? I have a similar problem except I need to enter *1.
When I do, it seems like the panel is rebooted or something as the physical keypad loses its "Armed" LED for about 10 seconds and then comes back on.

There does appear to be a function in the library for night arming, but when I use it my alarm does not behave the same as "Stay" -> "*1"

**Comment by bourgeois** [comment]
No, I never managed to make this work 😞

Let me know if you ever find something!

**Comment by Dilbert66** [comment]
You might need to disarm first then go to night mode. Also, on my system you also need to send the access code after the *9 to go to night mode as an additional dsc.write (code)

**Comment by duanemck** [comment]
I see in version 2.0 there is now support for *1 to switch between bypass modes. However I'm not sure how to use it. I setup my sketch to arm Stay (write 's') then wrote '*1' after a dsc.loop(). The keypad suddenly changes to say "Unavailable" but then seems to recover.

Is there a specific way to use this new feature? @taligentx

**Comment by taligentx** [comment]
I've updated the Homebridge sketch and the library code in the `develop` branch to support switching the armed modes while armed - switching between armed stay/away and night arm is immediate, while switching between armed stay and away triggers the exit delay.

Instead of using `*1` and `*9` to switch modes, the sketch is using the keypad shortcut keys for each mode - tested on the PC1864, let's see how this works out for the other panels.

**Reply by duanemck** [reply]
Thank you, I'll go have a look in `develop`. And thanks so much for all the hard work put into this library!

**Reply by duanemck** [reply]
I updated my code to use the `develop` branch and had a look at your samples.
It's definitely doing something on the panel as the keypad "armed" led changes from solid to flashing. But it's not behaving how `*1` does. I suspect the installer set it up to do something other than the normal night mode when pressing `*1`.

I'm going to have another go at sending `*1` manually while armed in Stay and see what happens.

If not, I'll see if I can dump the commands being sent when I do it from the keypad and perhaps you'll be able to use that to see what it's actually doing?
